geometry3d/volLocalMax.cpp

```diff
@@ -0,0 +1,144 @@
+
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+/**
+ * @file vol2obj.cpp
+ * @author Bertrand Kerautret (\c bertrand.kerautret@loria.fr)
+ *
+ * @date 2015/12/22
+ *
+ *
+ * This file is part of the project DGtal-Tools-Contrib of the DGtal library.
+ */
+
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/io/readers/GenericReader.h"
+#include "DGtal/io/boards/Board3D.h"
+#include "DGtal/io/DrawWithDisplay3DModifier.h"
+#include "DGtal/shapes/implicit/ImplicitBall.h"
+#include "DGTal/shapes/GaussDigitizer.h"
+
+
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+
+using namespace std;
+using namespace DGtal;
+using namespace Z3i;
+
+///////////////////////////////////////////////////////////////////////////////
+namespace po = boost::program_options;
+
+typedef ImageContainerBySTLVector<Z3i::Domain, unsigned char> Image3D;
+
+
+
+template<typename TImage>
+bool
+isLocalMax(const TImage &anImage, const Z3i::Point &aPoint, double aRadius){
+  typedef DGtal::ImplicitBall<DGtal::Z3i::Space> EuclideanBall;
+  typedef DGtal::GaussDigitizer<DGtal::Z3i::Space, EuclideanBall> DigitalShape;
+  EuclideanBall aBall (DGtal::Z3i::Point(aPoint[0], aPoint[1], aPoint[2]), aRadius);
+  unsigned int myVal = anImage(aPoint);
+  DigitalShape  gaussDig;
+  gaussDig.attach (aBall);
+  gaussDig.init(DGtal::Z3i::Point(aPoint[0]-(int)aRadius-1, aPoint[1]-(int)aRadius-1, aPoint[2]-(int)aRadius-1),
+                DGtal::Z3i::Point(aPoint[0]+(int)aRadius+1, aPoint[1]+(int)aRadius+1, aPoint[2]+(int)aRadius+1), 1);
+  DGtal::Z3i::Domain dom = gaussDig.getDomain();
+  for( DGtal::Z3i::Domain::ConstIterator it = dom.begin(); it!=dom.end(); it++){
+    if( *it != aPoint){
```

Comment:
if we remove the if it will return alway false (we have <=). but the point test can be moved inside the <= test to be a little more efficient ;)